src/components/App/index.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from "react";
+import { ThemeProvider } from "@mui/material";
+import CssBaseline from "@mui/material/CssBaseline";
+import { useAuthState } from "react-firebase-hooks/auth";
+import { getAuth } from "firebase/auth";
+import { Router, navigate, RouteComponentProps } from "@reach/router";
+import { IntlConfig, IntlProvider } from "react-intl";
+
+import Login from "../../views/Login";
+import theme from "../../theme";
+import { HomeView } from "../../views/Home";
+import { firebaseApp } from "../../index";
+
+type Translations = {
+  es: IntlConfig["messages"];
+  en: IntlConfig["messages"];
+};
+
+const translations: Translations = {
+  es: require("../../translations/es.json"),
+  en: require("../../translations/en.json"),
+};
+
+const navLang = navigator.language.split("-")[0];
+
+const lang = isTranslation(navLang) ? navLang : "en";
+
+function isTranslation(langugage?: string): langugage is keyof Translations {
+  if (!langugage) return false;
+  return Object.keys(translations).includes(langugage);
+}
+
+export const App = () => (
+  <IntlProvider locale={lang} messages={translations[lang]} defaultLocale="en">
+    <CssBaseline />
+    <ThemeProvider theme={theme}>
+      <Router>
+        <Login path="/auth/login" />
+        <Authorized path="/*" />
+        {/* <Signup path="/auth/signup" /> */}
+        {/* <ResetPassword path="/auth/reset-password" /> */}
+        {/* <Authorized path="/*" /> */}
+      </Router>
+    </ThemeProvider>
+  </IntlProvider>
+);
+
+const Authorized = ({ location }: RouteComponentProps) => {
```

Comment:
Used the [Reach Router Types](https://reach.tech/router/typescript) here, instead of manually creating them